runtime/moonbeam/src/lib.rs

```diff
@@ -1367,6 +1384,21 @@ impl pallet_parameters::Config for Runtime {
 	type WeightInfo = moonbeam_weights::pallet_parameters::WeightInfo<Runtime>;
 }
 
+impl pallet_multiblock_migrations::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type Migrations = pallet_identity::migration::v2::LazyMigrationV1ToV2<Runtime>;
+	// Benchmarks need mocked migrations to guarantee that they succeed.
+	#[cfg(feature = "runtime-benchmarks")]
+	type Migrations = pallet_multiblock_migrations::mock_helpers::MockedMigrations;
+	type CursorMaxLen = ConstU32<65_536>;
+	type IdentifierMaxLen = ConstU32<256>;
+	type MigrationStatusHandler = ();
+	type FailedMigrationHandler = frame_support::migrations::FreezeChainOnFailedMigration;
```

Comment:
We should not use FreezeChainOnFailedMigration in production, according to the doc of this type:

> /// Do now allow any transactions to be processed after a runtime upgrade failed.
> ///
> /// This is **not a sane default**, since it prevents governance intervention.

We have to create our own implementation of FailedMigrationHandler that trigger our maintenance mode.

Reply:
Sorry I didn't explain how critical it is: If a multiblock migration fail `FreezeChainOnFailedMigration` will disable the governance, then we will not be able to perform any runtime upgrade, it's a very critical issue

Reply:
Drafted a first solution, please let me know what you think! https://github.com/moonbeam-foundation/moonbeam/pull/3224/commits/d50b3044552a2910142b42647f76efcf80e4a20d